corehq/apps/reports/view_helpers.py

```diff
@@ -23,15 +23,13 @@ def case_hierarchy_context(case, timezone=None):
         for idx in case.live_indices:
             try:
                 parent_case = idx.referenced_case
-                parent_case.index_info = _index_to_context(idx, is_ancestor=True)
-                parent_cases.append(parent_case)
             except ResourceNotFound:
                 parent_cases.append(None)
-        for parent_case in parent_cases:
-            if parent_case:
-                last_parent_id = parent_case.case_id
-            else:
                 last_parent_id = None
+            else:
+                parent_case.index_info = _index_to_context(idx, is_ancestor=True)
+                parent_cases.append(parent_case)
+                last_parent_id = parent_case.case_id
 
         for c in descendant_case_list:
```

Comment:
This whole loop could be skipped if `parent_cases[-1] is None`, and if that is done then `last_parent_id = parent_cases[-1].case_id` could be moved inside the `if` block and removed from the last line of the suggestion above.